react/src/Groups/components/CreateGroupModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from "react";
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    TextField,
+} from "@material-ui/core";
+import { useSnackbar } from "notistack";
+import { Group } from "../../typings";
+
+interface CreateGroupModalProps {
+    open: boolean;
+    onClose: () => void;
+}
+
+export default function CreateGroupModal(props: CreateGroupModalProps) {
+    const emptyGroup = { group_code: "", group_name: "" };
+    const [group, setGroup] = useState<Group>(emptyGroup);
+    const { enqueueSnackbar } = useSnackbar();
+
+    function onClose() {
+        setGroup(emptyGroup);
+        props.onClose();
+    }
+
+    async function submit(e: React.FormEvent) {
+        e.preventDefault();
+
+        const response = await fetch("/api/groups", {
+            method: "POST",
+            credentials: "same-origin",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify(group),
```

Comment:
I don't know if group codes are case-sensitive as far as MinIO is concerned, but since the codes are converted to upper-case in the group boxes, I think they should be made lower-case before sending.